zerver/lib/user_groups.py

```diff
@@ -84,15 +84,15 @@ def has_user_group_access(
 
 def access_user_group_by_id(
     user_group_id: int, user_profile: UserProfile, *, for_read: bool
-) -> UserGroup:
+) -> NamedUserGroup:
```

Comment:
I guess we won't need this function for `UserGroup` objects in future, so did change it for now to only consider `NamedUserGroup`.

Reply:
Yep, sounds correct!